pkg/v1/partial/configlayer_test.go

```diff
@@ -0,0 +1,97 @@
+package partial
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+type testUIC struct {
+	configFile []byte
+}
+
+func (t testUIC) RawConfigFile() ([]byte, error) {
+	return t.configFile, nil
+}
+
+func (t testUIC) MediaType() (types.MediaType, error) {
+	return "", nil
+}
+
+func (t testUIC) LayerByDiffID(h v1.Hash) (UncompressedLayer, error) {
+	return nil, fmt.Errorf("no layer by diff ID %v", h)
+}
+
+func TestUncompressedLayerByDigest(t *testing.T) {
+	uic := testUIC{
+		configFile: []byte("{}"),
+	}
+	uie := &uncompressedImageExtender{
+		UncompressedImageCore: uic,
+	}
+	testLayerByDigestForConfigHash(t, uie)
+}
+
+type testCIC struct {
+	configFile []byte
+}
+
+func (t testCIC) RawConfigFile() ([]byte, error) {
+	return t.configFile, nil
+}
+
+func (t testCIC) MediaType() (types.MediaType, error) {
+	return "", nil
+}
+
+func (testCIC) RawManifest() ([]byte, error) {
+	return nil, nil
+}
+
+func (t testCIC) LayerByDigest(h v1.Hash) (CompressedLayer, error) {
+	return nil, fmt.Errorf("no layer by diff ID %v", h)
+}
+
+func TestCompressedLayersByDigest(t *testing.T) {
```

Comment:
Can these be condensed into a table-driven test?

```
for _, image := range []v1.Image {
  &compressedImageExtender{
    CompressedImageCore: testCIC{
      configFile: []byte("{}"),
    },
}, {
  &uncompressedImageExtender{
    UncompressedImageCore: testUIC{
      configFile: []byte("{}"),
    },
  },
}} {
  hash, err := image.ConfigName()
  ... and the rest of testLayerByDigestForConfigHash ...
}
```

Reply:
Absolutely! I'm keeping the definition of the cases separate from the loop, if that's okay. If it's idiomatic otherwise, I'm happy to embed the table into the `for`.